skbase/base/_base.py

```diff
@@ -1323,19 +1324,13 @@ def get_fitted_params(self, deep=True):
         if not deep:
             return fitted_params
 
-        def sh(x):
-            """Shorthand to remove all underscores at end of a string."""
-            if x.endswith("_"):
-                return sh(x[:-1])
-            else:
-                return x
-
         # add all nested parameters from components that are skbase BaseEstimator
         c_dict = self._components()
         for c, comp in c_dict.items():
             if isinstance(comp, BaseEstimator) and comp._is_fitted:
                 c_f_params = comp.get_fitted_params(deep=deep)
-                c_f_params = {f"{sh(c)}__{k}": v for k, v in c_f_params.items()}
+                c = c.strip("_")
```

Comment:
two questions about this:

* this is not part of #281, is that intentional? This is not about consistent use of "estimator" either.
* this is not a refactor but a change in logic, hence potentially breaking dependent code. `sh` removed only a single trailing underscore, whereas `strip` removes both an arbitrary amount of trailing underscores, as well as leading underscores - the latter could be a consequential, breaking change, even in `sktime` I remember examples of things like `_sthsth_` as attribute names

Irrespective of what we think the "should state" should be, we should not make breaking changes without deprecation.

Reply:
Hey @fkiraly this was just some housecleanning indeed and not related to anything else. You are correct about the removal of leading underscores! That's my bad! I should've used `rstrip()` to only strip trailing underscores. Btw, the `sh()` helper also strips arbitrary trailing underscores (it uses recursion).

```pycon
>>> def sh(x):
...     """Shorthand to remove all underscores at end of a string."""
...    if x.endswith("_"):
...        return sh(x[:-1])
...    else:
...        return x    
>>> a = "_foo__bar____"
>>> sh(a)
'_foo__bar'
>>> a.rstrip("_")
'_foo__bar'
>>> sh(a) == a.rstrip("_")
True
>>> b = "___foo__bar"
>>> sh(b) == b.rstrip("_")
True
```

 Would you be okay with keeping the change but fixing it by using the `rstrip` method? The only reason I did it was because `sh()` is kinda reinventing the wheel since `rstrip` does exactly the same and seems a bit more idiomatic.
 
 I don't mind either way! 👍

Reply:
Just pushed a commit in case...

Reply:
> Would you be okay with keeping the change but fixing it by using the `rstrip` method?

Yes - that is equivalent, up to and including the failure if the input is not a `str`. So it would be a refactor, and an improvement.